marshmallow_utils/permissions.py

```diff
@@ -37,14 +38,17 @@ def _permissions_filter_load(self, data, **kwargs):
     @post_dump
     def _permissions_filter_dump(self, data, **kwargs):
         field_permission_check = self.context.get("field_permission_check")
-        # NOTE: we make a copy since we don't want to modify the data when
-        # deleting keys
-        # TODO (Alex): see if actually needed...
-        # data = deepcopy(data)
+        # Initialize "request" permissions cache
+        if not hasattr(request, "permissions"):
+            request.permissions_cache = dict()
```

Comment:
I would add `_` in front and make it more unique, just in case:
```suggestion
            request._field_permissions_cache = dict()
```